src/test/kotlin/org/jitsi/utils/TemplatedUrlTest.kt

```diff
@@ -0,0 +1,95 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.utils
+
+import io.kotest.assertions.throwables.shouldThrow
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.shouldBe
+import java.net.URI
+import java.net.URISyntaxException
+
+class TemplatedUrlTest : ShouldSpec({
+    context("TemplatedUrl") {
+        should("resolve a simple template") {
+            val template = "https://example.com/{{path}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("path", "test")
+
+            templatedUrl.resolve() shouldBe URI("https://example.com/test")
+        }
+
+        should("resolve a template with multiple keys") {
+            val template = "wss://{{host}}/{{path}}?param={{param}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("host", "example.com")
+            templatedUrl.set("path", "api/resource")
+            templatedUrl.set("param", "value")
+
+            templatedUrl.resolve() shouldBe URI("wss://example.com/api/resource?param=value")
+        }
+
+        should("resolve a template with multiple keys when they are not set()") {
+            val template = "wss://{{host}}/{{path}}?param={{param}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("host", "example.com")
+            templatedUrl.set("path", "api/resource")
+            templatedUrl.set("param", "value")
+
+            templatedUrl.resolve(
+                mapOf("host" to "example.com", "path" to "api/resource", "param" to "value")
+            ) shouldBe URI("wss://example.com/api/resource?param=value")
+        }
+
+        should("resolve with a new key-value pair without saving it") {
+            val template = "https://example.com/{{path}}?param={{param}}"
+            val templatedUrl = TemplatedUrl(template)
+            templatedUrl.set("path", "api")
+
+            // Resolve with temporary param value
+            templatedUrl.resolve("param", "temp") shouldBe URI("https://example.com/api?param=temp")
+
+            // Original template should not have the param saved
+            templatedUrl.set("param", "permanent")
+            templatedUrl.resolve() shouldBe URI("https://example.com/api?param=permanent")
+        }
+
+        should("throw an exception when not all requiredKeys are set") {
+            val template = "https://example.com/{{path}}?param={{param}}"
+            val templatedUrl = TemplatedUrl(template, requiredKeys = setOf("param"))
+            templatedUrl.set("path", "api")
+
+            shouldThrow<IllegalArgumentException> {
+                templatedUrl.resolve()
+            }
+        }
+        should("throw an exception when the URI is invalid") {
+            val templatedUrl = TemplatedUrl("https://example.com/{{}}")
```

Comment:
I take it this is relying on the fact than an unquoted `{` or `}` isn't valid anywhere in a URI?